File: apps/workshop-java/workshop/src/main/java/com/theagilemonkeys/workshop/config/SemanticSearchConfiguration.java
package com.theagilemonkeys.workshop.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConfigurationProperties(prefix = "semanticsearch")
public class SemanticSearchConfiguration {
    private String openaiKey;
    private String pineconeKey;
    private String pineconeUrl;

    public String getOpenaiKey() {
        return openaiKey;
    }

    public void setOpenaiKey(String openaiKey) {
        this.openaiKey = openaiKey;
    }

    public String getPineconeKey() {
        return pineconeKey;
    }

    public void setPineconeKey(String pineconeKey) {
        this.pineconeKey = pineconeKey;
    }

    public String getPineconeUrl() {
        return pineconeUrl;
    }

    public void setPineconeUrl(String pineconeUrl) {
        this.pineconeUrl = pineconeUrl;
    }
}

File: workshop-kotlin/src/main/kotlin/com/theagilemonkeys/workshop/config/SemanticSearchConfiguration.kt
package com.theagilemonkeys.workshop.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@ConfigurationProperties(prefix = "semanticsearch")
data class SemanticSearchConfiguration(
    var openaiKey: String = "",
    var pineconeKey: String = "",
    var pineconeUrl: String = "",
    var pineconeNamespace: String = ""
)

@Configuration
@EnableConfigurationProperties(SemanticSearchConfiguration::class)
class AppConfiguration {
    @Bean
    fun semanticSearchConfiguration(): SemanticSearchConfiguration {
        return SemanticSearchConfiguration()
    }
}
